source/common/tls/utility.cc

```diff
@@ -384,6 +384,24 @@ absl::optional<uint32_t> Utility::getDaysUntilExpiration(const X509* cert,
   return absl::nullopt;
 }
 
+std::vector<std::string> Utility::getCertificateExtensionOids(X509& cert) {
+  std::vector<std::string> extension_oids;
+
+  int count = X509_get_ext_count(&cert);
+  for (int pos = 0; pos < count; pos++) {
+    X509_EXTENSION* extension = X509_get_ext(&cert, pos);
+    RELEASE_ASSERT(extension != nullptr, "");
+
+    char oid[256];
+    int len = OBJ_obj2txt(oid, sizeof(oid), X509_EXTENSION_get_object(extension),
+                          1 /* always_return_oid */);
+    if (len > 0) {
```

Comment:
if `len > sizeof(oid)` there was an overflow and the value is truncated. Should that be handled specially?